nmdc_automation/workflow_automation/models.py

```diff
@@ -0,0 +1,199 @@
+
+    @property
+    def id(self):
+        return self.process.id
+
+    @property
+    def type(self):
+        return self.process.type
+
+    @property
+    def name(self):
+        return self.process.name
+
+    @property
+    def has_input(self):
+        return self.process.has_input
+
+    @property
+    def has_output(self):
+        return self.process.has_output
+
+    @property
+    def git_url(self):
+        """ workflow executions have a git_url field, data generations do not"""
+        default_url = "http://github.com/microbiomedata"
+        return getattr(self.process, "git_url", default_url)
+
+    @property
+    def version(self):
+        """ workflow executions have a version field, data generations do not"""
+        return getattr(self.process, "version", None)
+
+    @property
+    def analyte_category(self):
+        """ data generations have an analyte_category field, workflow executions do not"""
+        return getattr(self.process, "analyte_category", None)
+
+    @property
+    def was_informed_by(self):
+        """ workflow executions have a was_informed_by field, data generations get set to their own id"""
+        return getattr(self.process, "was_informed_by", self.id)
+
+
+class DataObject(nmdc.DataObject):
+    """
+    Extends the NMDC DataObject dataclass with additional methods for serialization.
+    """
+    def __init__(self, **record):
+        """ Initialize the object from a dictionary """
+        # _id is a MongoDB field that makes the parent class fail to initialize
+        record.pop("_id", None)
+        super().__init__(**record)
+
+    def as_dict(self):
+        """ Return the object as a dictionary, excluding None values, empty lists, and data_object_type as a string """
+        return {
+            key: value
+            for key, value in self.__dict__.items()
+            if not key.startswith("_") and value
+        } | {"data_object_type": self.data_object_type}
+
+    @property
+    def data_object_type(self):
+        """ Return the data object type as a string """
+        if isinstance(self._data_object_type, FileTypeEnum):
+            return self._data_object_type.code.text
+        return str(self._data_object_type)
+
+    @data_object_type.setter
+    def data_object_type(self, value):
+        """ Set the data object type from a string or FileTypeEnum """
+        if isinstance(value, FileTypeEnum):
+            self._data_object_type = value
+        else:
+            self._data_object_type = FileTypeEnum(value)
+
+
+@dataclass
+class WorkflowConfig:
+    """ Configuration for a workflow execution """
+    # Sequencing workflows only have these fields
+    name: str
+    collection: str
+    enabled: bool
+    analyte_category: str
+    filter_output_objects: List[str]
+    # TODO should type be optional?
+    type: Optional[str] = None
+
+    # workflow repository information
+    git_repo: Optional[str] = None
```

Comment:
git_repo is not an optional slot for this class.

Reply:
Sequencing workflows don't have a `git_repo`

Reply:
My strong preference is to remove Sequencing Interleaved and Sequencing Noninterleaved from the configs so this isn't an issue. This would also allow us to make version required on our end even though it isn't required by the schema currently. Workaround option is to add git_url: https://github.com/microbiomedata/RawSequencingData to Sequencing Interleaved and Sequencing Noninterleaved, that is what the existing mongo records use. This is a repo Shane made to deal with the requirement on the schema end.

Reply:
Why was this comment resolved?